### PR TITLE
Fix transformation matrix computation

### PR DESCRIPTION
**Pull request by deathly809**

### CODE REVIEW

File: resources/models/tree.dae [hunk 1]
```diff
@@ -524,7 +524,7 @@
                         0.085229 -0.994523 0.0604959
                         -0.970396 -0.0980064 -0.220741
                         -0.0709067 0.988553 -0.133176
-                        0.968858 0.100014 0.22652
```
Comment (deathly809) [inline resources/models/tree.dae]: Whoops

File: src/Resources/Mesh.cpp [hunk 2]
```diff
@@ -1,25 +1,48 @@
...
 
-#include <ResourceManager.hpp>
+#include <sstream>
 
-#include <Logger.hpp>
+#include <glError.hpp>
 
 using namespace models;
 
+std::string ToString(glm::mat4& mat) {
```
Comment (deathly809) [inline src/Resources/Mesh.cpp:17]: We should make a utility class for all these "ToString" methods.

File: resources/shaders/shader.frag [hunk 3]
```diff
@@ -1,26 +1,30 @@
...
 
 // output
-out vec4 color;
+out vec4 OutColor;
 
 // white light
 vec3 lightColor = vec3(1.0, 1.0, 1.0);
 
+// Object color
+vec4 objectColor = vec4(1.0, 1.0, 1.0, 1.0);
```
Comment (gabloe) [inline resources/shaders/shader.frag:18]: Shouldnt this come from the model?

**Review by gabloe — APPROVED** [review]

File: src/Resources/Model.cpp [hunk 4]
```diff
@@ -21,8 +21,21 @@ void Model::Load(std::string fileName) {
 
   this->path = fileName;
 
+  // Scale it back up
   logging::Logger::LogDebug("Processing root node");
-  ProcessNode(scene->mRootNode, scene);
+  {
+    auto scale = scene->mRootNode->mTransformation[0][0] == 0.0
```
Comment (deathly809) [inline src/Resources/Model.cpp:27]: I am not sure what is going on, but whenever we render the models they are not rendered to size. When compared to assimp_view they are much much smaller. More investigation into how these models are supposed to be loaded is needed.

File: src/TerrainGenerator.cpp [hunk 5]
```diff
@@ -33,29 +33,31 @@ TerrainGenerator::TerrainGenerator(config::ConfigReader& configReader)
...
     modelPath = asset::Asset::Instance().MODELS_DIR + "/" + modelName;
-    logging::Logger::LogInfo("Overriding default model vale: " + modelPath);
+    logging::Logger::LogInfo("Overriding default model value: " + modelPath);
   }
 
   if (configReader.ContainsKey("vertexShader")) {
     std::string vertexShaderName = configReader.ReadString("vertexShader");
-    vertexShaderPath = asset::Asset::Instance().SHADERS_DIR + "/" + vertexShaderName;
+    vertexShaderPath =
+        asset::Asset::Instance().SHADERS_DIR + "/" + vertexShaderName;
```
Comment (deathly809) [inline src/TerrainGenerator.cpp:46]: In a follow-up PR change all these string paths to std::filesystem::path types

**Pull request merged**